src/main/java/teammates/googleSendgridJava/Sendgrid.java

```diff
@@ -0,0 +1,509 @@
+/**
+ * Imported from https://github.com/sendgrid/sendgrid-google-java
+ * 
+ * The link to the Github repo is provided in the documentation for Bulk Mail With Analytics
+ * via Sendgrid for Java: https://cloud.google.com/appengine/docs/java/mail/sendgrid
+ */
+
+package teammates.googleSendgridJava;
+
+import java.net.HttpURLConnection;
+import java.util.*;
+import java.io.IOException;
+import java.util.Iterator;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+
+import com.google.appengine.labs.repackaged.org.json.JSONException;
+import com.google.appengine.labs.repackaged.org.json.JSONObject;
+import com.google.appengine.labs.repackaged.org.json.JSONArray;
+
+public class Sendgrid {
```

Comment:
This class represents an Email message compatible with SendGrid or the SendGrid API for sending emails? If it is the former, the class name does not represent the purpose of the class. If it is both, then the class breaks the Single Responsibility principle.  See if you can split it up.